apps/www/components/headers/navigationHeader/darkModeSwitcher/DarkModeSwitcher.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { useCallback, useEffect, useState } from 'react';
+
+import styles from './darkModeSwitcher.module.scss';
+
+const getInitialColorMode = () => {
+  if (typeof window === 'undefined') {
+    return 'light';
+  }
+
+  const persistedColorPreference = window.localStorage.getItem('color-mode');
+
+  if (persistedColorPreference === 'dark' || persistedColorPreference === 'light') {
+    return persistedColorPreference;
+  }
+
+  const mql = window.matchMedia('(prefers-color-scheme: dark)');
+  const hasMediaQueryPreference = typeof mql.matches === 'boolean';
+  if (hasMediaQueryPreference) {
+    return mql.matches ? 'dark' : 'light';
+  }
+
+  return 'light';
+};
+
+const DarkModeSwitcher = React.memo(() => {
+  const [colorMode, setColorMode] = useState<'light' | 'dark' | undefined>();
```

Comment:
A tu nie powinno być
```suggestion
  const [colorMode, setColorMode] = useState<'light' | 'dark' | undefined>(() => getInitialColorMode());
```
?

Reply:
Masz racje, poprawiam